activerecord/lib/active_record/connection_adapters/abstract_mysql_adapter.rb

```diff
@@ -137,6 +137,11 @@ def supports_insert_on_duplicate_update?
         true
       end
 
+      def field_ordered_value(column, values)
+        field = Arel::Nodes::NamedFunction.new("FIELD", [column, values.reverse])
+        Arel::Nodes::Descending.new(field)
```

Comment:
Could this avoid reversing `values` with `Arel::Nodes::Ascending`?

Reply:
I can't 100% remember why I did this, but it has to do with mysql putting nulls first or last, so it needed to be this way. I'll see if I can find out why.